demos/sample_azure_iot/sample_azure_iot.c

```diff
@@ -0,0 +1,625 @@
+
+        #ifdef democonfigDEVICE_SYMMETRIC_KEY
+            xResult = AzureIoTProvisioningClient_SetSymmetricKey( &xAzureIoTProvisioningClient,
+                                                                  ( const uint8_t * ) democonfigDEVICE_SYMMETRIC_KEY,
+                                                                  sizeof( democonfigDEVICE_SYMMETRIC_KEY ) - 1,
+                                                                  Crypto_HMAC );
+            configASSERT( xResult == eAzureIoTProvisioningSuccess );
+        #endif // democonfigDEVICE_SYMMETRIC_KEY
+
+        do
+        {
+            xResult = AzureIoTProvisioningClient_Register( &xAzureIoTProvisioningClient,
+                                                           sampleazureiotProvisioning_Registration_TIMEOUT_MS );
+        } while( xResult == eAzureIoTProvisioningPending );
+
+        configASSERT( xResult == eAzureIoTProvisioningSuccess );
+
+        xResult = AzureIoTProvisioningClient_GetDeviceAndHub( &xAzureIoTProvisioningClient,
+                                                              ucSampleIotHubHostname, &ucSamplepIothubHostnameLength,
+                                                              ucSampleIotHubDeviceId, &ucSamplepIothubDeviceIdLength );
+        configASSERT( xResult == eAzureIoTProvisioningSuccess );
+
+        AzureIoTProvisioningClient_Deinit( &xAzureIoTProvisioningClient );
+
+        /* Close the network connection.  */
+        TLS_Socket_Disconnect( &xNetworkContext );
+
+        *ppucIothubHostname = ucSampleIotHubHostname;
+        *pulIothubHostnameLength = ucSamplepIothubHostnameLength;
+        *ppucIothubDeviceId = ucSampleIotHubDeviceId;
+        *pulIothubDeviceIdLength = ucSamplepIothubDeviceIdLength;
+
+        return 0;
+    }
+
+#endif // democonfigENABLE_DPS_SAMPLE
+/*-----------------------------------------------------------*/
+
+/**
+* @brief Connect to server with backoff retries.
+*/
+static TlsTransportStatus_t prvConnectToServerWithBackoffRetries( const char * pcHostName,
+                                                                  uint32_t port,
+                                                                  NetworkCredentials_t * pxNetworkCredentials,
+                                                                  NetworkContext_t * pxNetworkContext )
+{
+    TlsTransportStatus_t xNetworkStatus;
+    BackoffAlgorithmStatus_t xBackoffAlgStatus = BackoffAlgorithmSuccess;
+    BackoffAlgorithmContext_t xReconnectParams;
+    uint16_t usNextRetryBackOff = 0U;
+
+    /* Initialize reconnect attempts and interval. */
+    BackoffAlgorithm_InitializeParams( &xReconnectParams,
+                                       sampleazureiotRETRY_BACKOFF_BASE_MS,
+                                       sampleazureiotRETRY_MAX_BACKOFF_DELAY_MS,
+                                       sampleazureiotRETRY_MAX_ATTEMPTS );
+
+    /* Attempt to connect to MQTT broker. If connection fails, retry after
```

Comment:
Simialr feedback about not using words "MQTT broker" here & other places in this it may come up.